packages/uikit-react-native/src/domain/groupChannel/module/moduleContext.tsx

```diff
@@ -48,6 +60,11 @@ export const GroupChannelContextsProvider: GroupChannelModule['Provider'] = ({
   const [typingUsers, setTypingUsers] = useState<SendbirdUser[]>([]);
   const [messageToEdit, setMessageToEdit] = useState<SendbirdUserMessage | SendbirdFileMessage>();
   const [messageToReply, setMessageToReply] = useState<SendbirdUserMessage | SendbirdFileMessage>();
+  const [scrollToMessage, __internalSetScrollToMessageFunc] = useState<GroupChannelScrollToMessageFunc>(() => () => {
+    Logger.error(
+      'You should render `src/domain/groupChannel/component/GroupChannelMessageList.tsx` component first to use scrollToMessage.',
```

Comment:
nit; when I saw the log message I couldn't get what it means exactly. Probably it's because I don't know much about the whole structures 😅 

Should we also mention where the component(GroupChannelMessageList) needs to be rendered too? i.e. GroupChannelMessageList should be a child of moduleContext / GroupChannelMessageList should be rendered somewhere independently..

Reply:
@AhyoungRyu RN 에서 스크롤을 하기 위해서는 FlatList 컴포넌트의 ref 를 통해서 직접 조작해야 하는데요

뎁스가 깊은곳에 있는 MessageList 컴포넌트에 forwardRef 와 useImperativeHandle 을 설정하고, 고객이 그것을 사용하게 열어주는게, 작업하는 쪽이나 사용하는 쪽이나 좋지 않은 방법이라고 생각을 해서
메세지 리스트 컴포넌트가 렌더링 된 이후에 context 에 set 해주고 그것을 사용하게끔 설계를 했었는데
리뷰 코멘트를 받아보니 이 부분이 좋지 않은 설계라는 생각이 들었습니다. 🤔

ref 를 상위 context 로 끌어올리고, context 에서 ref 를 조작하여 스크롤 하는 함수들을 생성하면
자연스럽게 말씀하신 부분이랑, 제가 고민한 커스터마이제이션/사용성 부분 또한 해결 가능할 것 같아서 이 방향으로 수정을 다시 해보았습니다.